File: source_funcs/go_transcoder/FIND_MINIMUM_DIFFERENCE_PAIR/FIND_MINIMUM_DIFFERENCE_PAIR_emit.go
package main

import "math"

//go:export f_gold
func f_gold(arr []int, n int) int {
	var diff int = math.MaxInt64
	for i := int(0); i < n-1; i++ {
		for j := int(i + 1); j < n; j++ {
			if math.Abs(float64(arr[i]-arr[j])) < float64(diff) {
				diff = int(math.Abs(float64(arr[i] - arr[j])))
			}
		}
	}
	return diff
}

File: rust_funcs/c_transcoder/FIND_MINIMUM_DIFFERENCE_PAIR/FIND_MINIMUM_DIFFERENCE_PAIR_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::collections::HashMap;


#[no_mangle]
fn f_gold(arr: [i32;2], n: i32) -> i32 {
    let mut diff = i32::MAX; 
    for i in 0..(n-1) as usize {
        for j in (i+1)..n as usize {
            diff = diff.min( (arr[i] - arr[j]).abs() );
        }
    }
    diff
}
